routes/subscriptions.js

```diff
@@ -13,15 +14,33 @@ function hidePrivateData(item) {
 
 router.post('/', function(req, res, next) {
   const family = req.user.family;
-  subscriptions.create(family, req.body).then((id) => {
-    debug('Subscription #%s has been created in database', id);
-
-    return subscriptions.show(family, id);
-  }).then((subscription) => {
-    res
-      .status(201)
-      .location(`${req.baseUrl}/${subscription.id}`)
-      .send(hidePrivateData(subscription));
+
+  const endpoint = req.body.subscription.endpoint;
+  subscriptions.findByEndpoint(family, endpoint).then((subscription) => {
+    // This endpoint is already in the DB
+    throw new DuplicateEndpointError( // will generate a status "409"
```

Comment:
Thanks for the reminder 😃 . I'd add "Conflict" in the code comment, for next readers.